core/lib/dal/src/blocks_dal.rs

```diff
@@ -557,17 +568,78 @@ impl BlocksDal<'_, '_> {
         Ok(())
     }
 
+    /// Inserts an unsealed L1 batch with some basic information (i.e. runtime related data is either
+    /// null or set to default value for the corresponding type).
     pub async fn insert_l1_batch(
+        &mut self,
+        number: L1BatchNumber,
+        timestamp: u64,
+        batch_fee_input: BatchFeeInput,
+    ) -> DalResult<()> {
+        sqlx::query!(
+            r#"
+            INSERT INTO
+                l1_batches (
+                    number,
+                    timestamp,
+                    l1_gas_price,
+                    l2_fair_gas_price,
+                    fair_pubdata_price,
+                    l1_tx_count,
+                    l2_tx_count,
+                    bloom,
+                    priority_ops_onchain_data,
+                    initial_bootloader_heap_content,
+                    used_contract_hashes,
+                    created_at,
+                    updated_at,
+                    is_sealed
+                )
+            VALUES
+                (
+                    $1,
+                    $2,
+                    $3,
+                    $4,
+                    $5,
+                    0,
+                    0,
+                    ''::bytea,
+                    '{}'::bytea[],
+                    '{}'::jsonb,
+                    '{}'::jsonb,
+                    NOW(),
+                    NOW(),
+                    FALSE
+                )
+            "#,
+            i64::from(number.0),
+            timestamp as i64,
+            batch_fee_input.l1_gas_price() as i64,
+            batch_fee_input.fair_l2_gas_price() as i64,
+            batch_fee_input.fair_pubdata_price() as i64,
```

Comment:
i64 is too small

Reply:
Don't we cap pubdata price at 10^15, so it should always fit in i64? I don't like that this invariant is very implicit but I don't see an easy way to avoid lossy conversions unless we want to invest some time into "custom" capped primitives

Reply:
yes, it's at the moment, but in the very next upgrade that is going to happen very soon this limit is increased to `2^64 - 1`. Can we use `numeric` type?

Reply:
Ah, didn't know that. We can try `numeric` type, sure